alpha/lib/model/LiveEntry.php

```diff
@@ -629,80 +642,80 @@ public function setMediaServer($index, $hostname, $applicationName = null)
-	{	
-		$server = $this->getFromCustomData("server-$index", LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
-		if($server && $server->getHostname() == $hostname)
-		{
-			$server = $this->removeFromCustomData("server-$index", LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
-			$this->setLastBroadcastEndTime(kApiCache::getTime());
-		}
-		
-		// Set media server with index #index to be stopped
-		// the entry->liveStatus() is calculated from the state of both servers
-		$this->setLiveStatus(LiveEntryStatus::STOPPED, $index);
-	}
-	
 	/**
 	 * @return bool true is list changed
 	 */
 	public function validateMediaServers()
 	{
 		$listChanged = false;
-		$kMediaServers = $this->getFromCustomData(null, LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS, array());
-		foreach($kMediaServers as $key => $kMediaServer)
+		$dbLiveEntryServerNodes = EntryServerNodePeer::retrieveByEntryId($this->id);
+		/* @var $dbLiveEntryServerNode LiveEntryServerNode */
+		foreach($dbLiveEntryServerNodes as $dbLiveEntryServerNode)
 		{
-			if(!$kMediaServer || ! $this->isCacheValid($kMediaServer))
+			if (! $this->isCacheValid($dbLiveEntryServerNode))
 			{
 				$listChanged = true;
-				KalturaLog::info("Removing media server [" . ($kMediaServer ? $kMediaServer->getHostname() : $key) . "]");
-				$this->removeFromCustomData($key, LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS);
+				// we need to set its live_status field if the supported line is deleted
+				$this->setLiveStatus(EntryServerNodeStatus::STOPPED, $dbLiveEntryServerNode->getServerType());
+				KalturaLog::info("Removing media server id".$dbLiveEntryServerNode->getServerNodeId());
+				$dbLiveEntryServerNode->delete();
 			}
 		}
-		
 		return $listChanged;
 	}
-	
+
 	public function getLiveStatus ($mediaServerIndex = null)
 	{
 		if ($mediaServerIndex != null){
 			return $this->getLiveStatusInternal($mediaServerIndex);
 		}
-		return LiveEntry::maxLiveEntryStatus($this->getLiveStatusInternal(MediaServerIndex::PRIMARY), $this->getLiveStatusInternal(MediaServerIndex::SECONDARY));
+		return LiveEntry::maxLiveEntryStatus($this->getLiveStatusInternal(EntryServerNodeType::LIVE_PRIMARY), $this->getLiveStatusInternal(EntryServerNodeType::LIVE_BACKUP));
 	}
-
 	private function getLiveStatusInternal ($mediaServerIndex)
 	{
-		return $this->getFromCustomData('live_status_'.$mediaServerIndex, null, LiveEntryStatus::STOPPED);
+		return $this->getFromCustomData('live_status_'.$mediaServerIndex, null, EntryServerNodeStatus::STOPPED);
 	}
-
 	public function setLiveStatus ($v, $mediaServerIndex)
 	{
 		$this->putInCustomData('live_status_'.$mediaServerIndex, $v);
 	}
-	
+
 	/**
-	 * @return array<kLiveMediaServer>
+	 * @return array<LiveEntryServerNode>
 	 */
-	public function getMediaServers()
+	public function getLiveEntryServerNodes()
 	{
-		return $this->getFromCustomData(null, LiveEntry::CUSTOM_DATA_NAMESPACE_MEDIA_SERVERS, array());
+		return EntryServerNodePeer::retrieveByEntryId($this->getId());
```

Comment:
Here you should return only EntryServerNodePeer that hold entries in status PLAYABLE.